Taskfile.yml

```diff
@@ -0,0 +1,93 @@
+  default:
+    cmds:
+      - task: list
+  list:
+    desc: list tasks
+    prefix: ' '
+    cmds:
+      - task --list --color=true --list-all
+  vars:
+    desc: variable output
+    cmds:
+      - |
+        echo -e "{{.light_gray}}=== ℹ {{ .reversed }} Variable Info from Task {{ .nocolor }} === "
+        echo -e "{{.light_gray}}MODEL_DIRECTORY          {{.nocolor}}: {{ .orange}}{{ .MODEL_DIRECTORY }} {{ .nocolor }}"
+        echo -e "{{.light_gray}}DOCKER_BUILDKIT          {{.nocolor}}: {{ .orange}}{{ .DOCKER_BUILDKIT }} {{ .nocolor }}"
+        echo -e "{{.light_gray}}UI                       {{.nocolor}}: {{ .orange}}{{ .UI }} {{ .nocolor }}"
+
+  ### Core Task ###
+  init:
+    desc: Download models, build initial docker-compose setup
+    cmds:
+    - task: download
+    - task: update
+  compose:build:
+    desc: setup tooling for project and download dependencies
+    cmds:
+      - |
+        docker compose --profile download up --build
+  compose:up:
+    desc: 'compose up with the current profile of: {{ .UI }}'
+    cmds:
+    - task: vars
+    - docker compose --profile {{ .UI }} up
+  update:
+    desc: run updates
+    interactive: true
+    cmds:
+    - task: vars
+    - |
+      docker compose --profile {{ .UI }} up --build
+    - task: download
+  download:
+    desc: get extras downloaded
+    cmds:
+    - |
+      echo '👉 This might take a few minutes. However, it will resume downloading if interuppted and ran again'
+      mkdir -p cache/models || true
+      curl --progress-bar -C - https://www.googleapis.com/storage/v1/b/aai-blog-files/o/sd-v1-4.ckpt?alt=media -o {{ .MODEL_DIRECTORY }}/model.ckpt
+      curl --progress-bar -C - https://github.com/TencentARC/GFPGAN/releases/download/v1.3.0/GFPGANv1.3.pth -o {{ .MODEL_DIRECTORY }}/GFPGANv1.3.pth
+      curl --progress-bar -C - https://github.com/xinntao/Real-ESRGAN/releases/download/v0.1.0/RealESRGAN_x4plus.pth -o {{ .MODEL_DIRECTORY }}/RealESRGAN_x4plus.pth
+      curl --progress-bar -C - https://github.com/xinntao/Real-ESRGAN/releases/download/v0.2.2.4/RealESRGAN_x4plus_anime_6B.pth -o {{ .MODEL_DIRECTORY }}/RealESRGAN_x4plus_anime_6B.pth
+      curl --progress-bar -C - https://github.com/xinntao/Real-ESRGAN/releases/download/v0.2.1/RealESRGAN_x2plus.pth -o {{ .MODEL_DIRECTORY }}/RealESRGAN_x2plus.pth
+      curl --progress-bar -C - https://heibox.uni-heidelberg.de/f/578df07c8fc04ffbadf3/?dl=1 -o {{ .MODEL_DIRECTORY }}/LDSR.ckpt
+      curl --progress-bar -C - https://heibox.uni-heidelberg.de/f/31a76b13ea27482981b4/?dl=1 -o {{ .MODEL_DIRECTORY }}/LDSR.yaml
+    status:
+      - test -f {{ .MODEL_DIRECTORY }}/model.ckpt
+      - test -f {{ .MODEL_DIRECTORY }}/GFPGANv1.3.pth
+      - test -f {{ .MODEL_DIRECTORY }}/RealESRGAN_x4plus.pth
+      - test -f {{ .MODEL_DIRECTORY }}/RealESRGAN_x4plus_anime_6B.pth
+      - test -f {{ .MODEL_DIRECTORY }}/RealESRGAN_x2plus.pth
+      - test -f {{ .MODEL_DIRECTORY }}/LDSR.ckpt
+      - test -f {{ .MODEL_DIRECTORY }}/LDSR.yaml
```

Comment:
Taskfile using `sh` a Go based shell interpreter that should run it all on windows just fine. The only catch to this was probably me forgetting mkdir, which might not. Otherwise, most of the commands there should work cross platform (which is why I use this over Make files).

Reply:
Thats cool! does it also have all of the basic utilities? stuff like `awk` `grep` `sed` etc?